### PR TITLE
Adds in a new chaplain antagonist item, the heartlight knife

### PR DESCRIPTION
**Pull request by Crackaduck**
[INPUT WANTED]
## About the PR 
This PR adds in a new chaplain antag item, costing 6 TC, a scroll that once read, replaces the readers heart and grants them the ability to summon a slightly less powerful Infiltrator dagger for 8 seconds with a cooldown of 30 seconds.
image
image
image
image
The name's could use some work however, I'm iffy on them and open to change

## Why's this needed? 
The chaplain has very little in terms of unique antag items, aside from ones that lean more into RP aspects of the role than damage, it gives medbay a good reason to practice heart surgery, and syndicate knives are just a fun item to use.

## Changelog

```changelog
(u)Carsontheking
(+)Adds In a new antag item for chaplains!
```

### CODE REVIEW

File: code/obj/item/organ_holder.dm [hunk 1]
```diff
@@ -1616,3 +1616,17 @@
...
+	name = "Sacred Heart"
+	desc = "Channel your faith in god into a knife to smite your enemies."
+	icon_state = "cyberkidney"
+	targeted = 0
+	cooldown = 30 SECONDS
+
+	cast(atom/target)
+		if (..())
+			return 1
+		boutput(holder.owner, "<span class='notice'>You use your [islist(linked_organ) ? "s" : ""] to conqure a knife from thin air.</span>")
```
Comment (DimWhat) [inline code/obj/item/organ_holder.dm:1630]: This is not a proper use nor spelling of "conquer"

**Review by DimWhat — COMMENTED** [review]
Couple description suggestions and some typos I found

File: code/obj/item/organ_holder.dm (hunk 1 above)
Comment (muzcl) [reply code/obj/item/organ_holder.dm:1630]: Probably meant to say 'conjure'

File: code/datums/syndicate_buylist.dm [hunk 2]
```diff
@@ -496,6 +496,16 @@ This is basically useless for anyone but miners.
 	vr_allowed = 0
 	can_buy = UPLINK_TRAITOR | UPLINK_SPY_THIEF
 
+/datum/syndicate_buylist/traitor/scroll
+	name = "Scroll of Sacred Heart's"
```
[the remark this answers is not in the document]
Comment (Zonespace27) [reply code/datums/syndicate_buylist.dm:500]:
```suggestion
	name = "Scroll of the Sacred Heart"
```
better

File: code/datums/syndicate_buylist.dm [hunk 3]
```diff
@@ -496,6 +496,16 @@ This is basically useless for anyone but miners.
...
+	item = /obj/item/device/sacred_heart_scroll
+	cost = 6
+	desc = "A scroll from ages past, sure to fill your heart with the lords truth."
+	job = list("Chaplain")
+	not_in_crates = 1
```
Comment (Zonespace27) [inline code/datums/syndicate_buylist.dm:505]: true/false defines

File: code/obj/item/device/sacred_heart.dm [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	desc = "A dusty old scroll containing essoteric knowlege from a time of cloak and dagger"
+	w_class = 2
+	is_syndicate = 1
+	var/implant = /obj/item/organ/heart/sacred
+	var/implants_available = HEART
+	var/list/parts_to_remove = list()
+	var/list/parts_to_add = list()
+
+	attack_self(mob/user as mob)
+		if (ishuman(user))
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:15]: early return

File: code/obj/item/device/sacred_heart.dm [hunk 5]
```diff
@@ -0,0 +1,61 @@
+#define HEART 1
+
+/obj/item/device/sacred_heart_scroll
+	name = "Sacred Heart Scroll"
+	icon_state = "sacred_heart_scroll-"
+	desc = "A dusty old scroll containing essoteric knowlege from a time of cloak and dagger"
+	w_class = 2
+	is_syndicate = 1
+	var/implant = /obj/item/organ/heart/sacred
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:9]: code doc comments

File: code/obj/item/device/sacred_heart.dm [hunk 6]
```diff
@@ -0,0 +1,61 @@
+#define HEART 1
+
+/obj/item/device/sacred_heart_scroll
+	name = "Sacred Heart Scroll"
+	icon_state = "sacred_heart_scroll-"
+	desc = "A dusty old scroll containing essoteric knowlege from a time of cloak and dagger"
+	w_class = 2
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:7]: are there not w_class defines

File: code/obj/item/device/sacred_heart.dm [hunk 7]
```diff
@@ -0,0 +1,61 @@
+#define HEART 1
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:1]: why this

File: code/obj/item/device/sacred_heart.dm [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+/obj/item/device/sacred_heart_scroll
+	name = "Sacred Heart Scroll"
+	icon_state = "sacred_heart_scroll-"
+	desc = "A dusty old scroll containing essoteric knowlege from a time of cloak and dagger"
+	w_class = 2
+	is_syndicate = 1
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:8]: true/false

File: code/obj/item/device/sacred_heart.dm [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+	var/implant = /obj/item/organ/heart/sacred
+	var/implants_available = HEART
+	var/list/parts_to_remove = list()
+	var/list/parts_to_add = list()
+
+	attack_self(mob/user as mob)
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:14]: as mob isn't needed

File: code/obj/item/device/sacred_heart.dm [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+		if (ishuman(user))
+			var/mob/living/carbon/human/H = user
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:16]: ideally don't use one letter vars

File: code/obj/item/device/sacred_heart.dm [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+			src.add_fingerprint(H)
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:17]: unneeded `src`

File: code/obj/item/device/sacred_heart.dm [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+				if ("Cancel")
+					return
+				if ("Proceed")
+					if (implants_available & HEART)
+						start_replace_heart(HEART, H)
+					else
+						user.show_text("This Scroll's Writings are Illegible.")
+
+
+	proc/start_replace_heart(var/target, var/mob/living/carbon/human/H)
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:28]: args don't need `var/`

File: code/obj/item/device/sacred_heart.dm [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+	var/implant = /obj/item/organ/heart/sacred
+	var/implants_available = HEART
+	var/list/parts_to_remove = list()
+	var/list/parts_to_add = list()
+
+	attack_self(mob/user as mob)
+		if (ishuman(user))
+			var/mob/living/carbon/human/H = user
+			src.add_fingerprint(H)
+			switch (alert("Would you like to read the [src] this can only be done once?",,"Proceed","Cancel"))
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:18]:
```suggestion
			switch (alert("Would you like to read [src]? This can only be done once.",,"Proceed","Cancel"))
```

File: code/obj/item/device/sacred_heart.dm [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+					return
+				if ("Proceed")
+					if (implants_available & HEART)
+						start_replace_heart(HEART, H)
+					else
+						user.show_text("This Scroll's Writings are Illegible.")
+
+
+	proc/start_replace_heart(var/target, var/mob/living/carbon/human/H)
+		parts_to_add += "heart"
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:29]: why add the heart here?

File: code/obj/item/device/sacred_heart.dm [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+		for(var/part_loc in parts_to_add)
+			var/obj/item/bodypart = null
+			bodypart = H.get_organ(part_loc)
+			if(bodypart)
+				parts_to_remove += part_loc
+		boutput(H, "<span class='alert'>This will take some time to read!</span>")
+		SPAWN(1 SECOND)
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:36]: you don't really need this at all

File: code/obj/item/device/sacred_heart.dm [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+					if (implants_available & HEART)
+						start_replace_heart(HEART, H)
+					else
+						user.show_text("This Scroll's Writings are Illegible.")
+
+
+	proc/start_replace_heart(var/target, var/mob/living/carbon/human/H)
+		parts_to_add += "heart"
+		for(var/part_loc in parts_to_add)
+			var/obj/item/bodypart = null
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:31]: compress lines 31 and 32

File: code/obj/item/device/sacred_heart.dm [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+			bodypart = H.get_organ(part_loc)
+			if(bodypart)
+				parts_to_remove += part_loc
+		boutput(H, "<span class='alert'>This will take some time to read!</span>")
+		SPAWN(1 SECOND)
+			playsound(H.loc, "sound/items/ocular_implanter_start.ogg", 50, 0, -1)
+			SETUP_GENERIC_ACTIONBAR(H, src, 10 SECONDS, /obj/item/device/sacred_heart_scroll/proc/end_replace_heart, list(target, H), src.icon, src.icon_state,"[src] Teaches you much about the faith.", null)
+
+	proc/end_replace_heart(var/target, var/mob/living/carbon/human/H)
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:40]: avoid one letter vars, no var/ needed

File: code/obj/item/device/sacred_heart.dm [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+		if(!H)
+			return
+		playsound(H.loc, "sound/items/ocular_implanter_end.ogg", 50, 0, -1)
+		var/turf/T = H.loc
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:44]: avoid one letter vars

File: code/obj/item/device/sacred_heart.dm [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+		for(var/part_loc in parts_to_remove)
+			if (T)
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:46]: put the if outside the for, cheaper

File: code/obj/item/device/sacred_heart.dm [hunk 20]
```diff
@@ -0,0 +1,61 @@
...
+				H.drop_organ(part_loc, T)
+				H.update_body()
+		for(var/part_loc in parts_to_add)
+			H.receive_organ(new implant, part_loc, 0, 1)
+			H.update_body()
+			implants_available = 0
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:52]: true/false defines

File: code/obj/item/device/sacred_heart.dm [hunk 21]
```diff
@@ -0,0 +1,61 @@
...
+			return
+		playsound(H.loc, "sound/items/ocular_implanter_end.ogg", 50, 0, -1)
+		var/turf/T = H.loc
+		for(var/part_loc in parts_to_remove)
+			if (T)
+				H.drop_organ(part_loc, T)
+				H.update_body()
+		for(var/part_loc in parts_to_add)
+			H.receive_organ(new implant, part_loc, 0, 1)
+			H.update_body()
```
Comment (Zonespace27) [inline code/obj/item/device/sacred_heart.dm:51]: call this outside of the for loop, update body isn't the cheapest proc

File: code/obj/item/organ_holder.dm [hunk 22]
```diff
@@ -1616,3 +1616,17 @@
 			src.icon_state = initial(src.icon_state)
 		else
 			src.icon_state = "[initial(src.icon_state)]_cd"
+
+/datum/targetable/organAbility/sacredheart
+	name = "Sacred Heart"
+	desc = "Channel your faith in god into a knife to smite your enemies."
+	icon_state = "cyberkidney"
+	targeted = 0
```
Comment (Zonespace27) [inline code/obj/item/organ_holder.dm:1624]: ture/false defines

File: code/obj/item/organ_holder.dm [hunk 23]
```diff
@@ -1616,3 +1616,17 @@
...
+	cooldown = 30 SECONDS
+
+	cast(atom/target)
+		if (..())
+			return 1
```
Comment (Zonespace27) [inline code/obj/item/organ_holder.dm:1629]: true/false defines

File: code/obj/item/organ_holder.dm (hunk 1 above)
Comment (Zonespace27) [inline code/obj/item/organ_holder.dm:1630]: in what case will this be a list

File: code/obj/item/organ_holder.dm [hunk 24]
```diff
@@ -1616,3 +1616,17 @@
...
+		boutput(holder.owner, "<span class='notice'>You use your [islist(linked_organ) ? "s" : ""] to conqure a knife from thin air.</span>")
+		var/obj/item/dagger/syndicate/chap_knife = new/obj/item/dagger/syndicate/chap_knife(get_turf(holder.owner))
```
Comment (Zonespace27) [inline code/obj/item/organ_holder.dm:1631]:
```suggestion
		var/obj/item/dagger/syndicate/chap_knife/chaplain_knife = new(get_turf(holder.owner))
```

File: code/obj/item/organ_holder.dm [hunk 25]
```diff
@@ -1616,3 +1616,17 @@
...
+		holder.owner.put_in_hand_or_drop(chap_knife)
```
Comment (Zonespace27) [inline code/obj/item/organ_holder.dm:1632]:
```suggestion
		holder.owner.put_in_hand_or_drop(chaplain_knife)
```

**Comment by Crackaduck** [comment]
Thank you all for the help, I'll be sure to start on it these changes later tonight. I'd also like to ask if anyone feels there ought to be any balance changes to this, be it the cooldown, damage delt, or time before deletion.

**Comment by DimWhat** [comment]
Just gonna note that you have a lot of unrelated sprite reversions in the DMI files you touched

**Comment by Crackaduck** [comment]
DimWhat, I Implemented most of the changes, thank you for the help, I did miss over one though, I'll get it at a later date I suppose

File: code/obj/item/device/sacred_heart.dm (hunk 11 above)
Comment (TobleroneSwordfish) [reply code/obj/item/device/sacred_heart.dm:17]: using `src.variable` is preferred I think

File: code/obj/item/organ_holder.dm (hunk 24 above)
Comment (Crackaduck) [reply code/obj/item/organ_holder.dm]: this one didn't work for whatever reason, it caused the knife to not spawn

File: code/obj/item/device/sacred_heart.dm (hunk 11 above)
Comment (Crackaduck) [reply code/obj/item/device/sacred_heart.dm:17]: I'm confused on this, is it fine or not wanted? I tried without a src and it didn't work so I just left it

File: code/obj/item/organ_holder.dm (hunk 1 above)
Comment (Crackaduck) [reply code/obj/item/organ_holder.dm]: probably none, It was just a holdover from other organ code

File: code/obj/item/organ_holder.dm [hunk 26]
```diff
@@ -1616,3 +1616,17 @@
 			src.icon_state = initial(src.icon_state)
 		else
 			src.icon_state = "[initial(src.icon_state)]_cd"
+
+/datum/targetable/organAbility/sacredheart
+	name = "Sacred Heart"
```
[the remark this answers is not in the document]
Comment (Crackaduck) [reply code/obj/item/organ_holder.dm]: I've renamed it to "Heartlight Blade" and I think that's probably not only better from a religious point of view, but more reflective and fun for the weapon and ability

**Comment by Crackaduck** [comment]
will need to go back and resprite the heart possibly, along with sound design, but code wise I think its about done. Not sure what's going on with maps though...

File: code/obj/item/device/sacred_heart.dm (hunk 7 above)
Comment (Crackaduck) [reply code/obj/item/device/sacred_heart.dm]: it won't work without it, idk why

**Comment by Crackaduck** [comment]
!merge_upstream

**Comment by Crackaduck** [comment]
well it works now!

File: code/obj/item/device/sacred_heart.dm [hunk 27]
```diff
@@ -0,0 +1,58 @@
+#define HEART 1 // I couldn't find a way to make this work without the define, my coding is very primitive however.
```
Comment (Azrun) [inline code/obj/item/device/sacred_heart.dm:1]: This comment adds no value

File: code/obj/item/device/sacred_heart.dm [hunk 28]
```diff
@@ -0,0 +1,58 @@
...
+				H.update_body()
+		for(var/part_loc in parts_to_add)
+			H.receive_organ(new implant, part_loc, 0, 1)
+			H.update_body()
+			implants_available = FALSE
+		boutput(H, "<span class='alert'><b>[pick("IT HURTS!", "OH GOD!", "JESUS FUCK!")]</b></span>")
+		bleed(H, 5, 5)
+		SPAWN(5 DECI SECOND)
+			H.emote("scream")
+		icon_state = "sacred_heart_scroll-U"
```
Comment (Azrun) [inline code/obj/item/device/sacred_heart.dm:54]: What is the benefit of having the scroll still exist? Evidence?

File: code/obj/item/device/sacred_heart.dm [hunk 29]
```diff
@@ -0,0 +1,58 @@
...
+		parts_to_add += "heart"
+		for(var/part_loc in parts_to_add)
+			var/obj/item/bodypart = null
+			bodypart = H.get_organ(part_loc)
+			if(bodypart)
+				parts_to_remove += part_loc
+		boutput(H, "<span class='alert'>You fill your heart with keen light !</span>")
+		end_replace_heart(HEART, H)
+
+	proc/end_replace_heart(var/target, var/mob/living/carbon/human/H)
```
Comment (Azrun) [inline code/obj/item/device/sacred_heart.dm:37]: `target` appears to be unused?

File: code/obj/item/device/sacred_heart.dm [hunk 30]
```diff
@@ -0,0 +1,58 @@
...
+				if ("Cancel")
+					return
+				if ("Proceed")
+					if (implants_available & HEART)
+						start_replace_heart(HEART, H)
+					else
+						user.show_text("This scroll's writings are illegible.")
+
+
+	proc/start_replace_heart(var/target, var/mob/living/carbon/human/H)
```
Comment (Azrun) [inline code/obj/item/device/sacred_heart.dm:27]: `target` appears to be unused?

File: code/obj/item/misc_weapons.dm [hunk 31]
```diff
@@ -538,6 +538,28 @@
...
+	desc = "A knife made out of pure light. Now thats what I call blind faith"
+	item_state = "chap_dagger"
+	force = 14
+	throwforce = 18
+	stamina_cost = 5
+	c_flags = EQUIPPED_WHILE_HELD
+
+	New()
+		..()
+		setProperty("movespeed", -0.4)
```
Comment (Azrun) [inline code/obj/item/misc_weapons.dm:553]: Why does this adjust move speed?

**Comment by Azrun** [comment]
Description seems misleading.  This is based on the infiltrator knife and not the syndicate dagger?

**Review by Azrun — COMMENTED** [review]
I'm not a weapons guy really but this feels a little weird to me.  Essentially a very slightly weaker infiltrators knife.  

Thematically I wish drawing the dagger out was more telegraphed and theatrical and/or traumatic in some way.  

Is this trying to be a magic divine thing?  Should it only work for chaplains or people with positive karma otherwise gib then?
Is it an artifact of some kind?  Maybe it would be something you plunge into your head instead?

tl;dr I'm not sure I get it...

File: code/obj/item/device/sacred_heart.dm (hunk 28 above)
Comment (Crackaduck) [reply code/obj/item/device/sacred_heart.dm:54]: ya mainly for evidence, and partially flavor. It makes it so you can't just use it in the corner of maint and never have to deal with at least disposing of it.

**Comment by Crackaduck** [comment]
> Reference

yes it is, I'll change it promptly

**Comment by Crackaduck** [comment]
> Description seems misleading. This is based on the infiltrator knife and not the syndicate dagger?

yes it is, my mistake I'll change it now.

**Comment by Crackaduck** [comment]
> Thematically I wish drawing the dagger out was more telegraphed and theatrical and/or traumatic in some way.
Would you say that adding in a audio que whenever the ability is used to tip people off would improve this?
 
> Is this trying to be a magic divine thing? Should it only work for chaplains or people with positive karma otherwise gib then? Is it an artifact of some kind? Maybe it would be something you plunge into your head instead?

Its trying to be divinely themed robotic implant of sorts, the idea being that it is made out of hardlight, though maybe rebranding it to be unstable hard-light would be more in line with the rest of the ss13 setting, given the lack of the technology and the fact it quickly disappears after initial use.

File: code/obj/item/misc_weapons.dm (hunk 31 above)
Comment (Crackaduck) [reply code/obj/item/misc_weapons.dm:553]: its a carry over from the syndicate infiltrator dagger, though nerfed comparatively. It helps give it utility and flavor, although if you believe it would be too powerful, it can be either nerfed or outright removed.

File: code/obj/item/device/sacred_heart.dm (hunk 27 above)
Comment (Crackaduck) [reply code/obj/item/device/sacred_heart.dm:1]: what exactly do you mean by that? the primitive part, or the entire comment? I added it there because zonespace previously commented on the define being un-needed.

File: code/obj/item/device/sacred_heart.dm (hunk 29 above)
Comment (Crackaduck) [reply code/obj/item/device/sacred_heart.dm:37]: The code doesn't work without it , SOMEHOW it messes up locating the target

File: code/obj/item/device/sacred_heart.dm (hunk 30 above)
Comment (Crackaduck) [reply code/obj/item/device/sacred_heart.dm:27]: closing this one as the comment above addresses the same issue.

**Comment by pali6** [comment]
I don't the the flavor of this quite fits. I'd probably lean more into it being a heart implant and away from the divine scroll stuff. But then it loses the connection with the chaplain so it doesn't fit either.

And overall the effect of the item doesn't sound particularly exciting to me at a glance.

**Comment by Crackaduck** [comment]
> I don't the the flavor of this quite fits. I'd probably lean more into it being a heart implant and away from the divine scroll stuff. But then it loses the connection with the chaplain so it doesn't fit either.

image
do you feel like something like this would fit more? more syndicate looking with the little flourish of the waxed paper seal on to give a more chaplain feel?

> And overall the effect of the item doesn't sound particularly exciting to me at a glance.
 :( (in all seriousness what do you think could be done to make it more chaplain orientated)

**Comment by Crackaduck** [comment]
so uh, any idea on the changes in response to the criticism?

**Comment by pali6** [comment]
I think I'm overall just not a fan of this, sorry. It doesn't feel interesting nor as a cohesive enough design to me.
